seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceReader.java

```diff
@@ -171,6 +171,9 @@ public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
                                                                 && record.offset()
                                                                         >= sourceSplit
                                                                                 .getEndOffset()) {
+                                                            // signal to the source that we have
```

Comment:
Thanks @SbloodyS for create this PR. Could you explain to us the purpose of move `context.signalNoMoreElement();` to here? And why this change can fix kafka source can not consumer all history data when use batch mode?

Reply:
Currently, In Batch mode The `pollNext` will only be executed once because `KafkaConsumer.poll` can only be consume limited data restricted by `max.poll.records` and its timeout. Moving it internally can enable `poolNext` to execute multiple times to consume data. @Hisoka-X

Reply:
Could you add test case to reproduce this bug?

Reply:
> Could you add test case to reproduce this bug? Like we can set `max.poll.records` to 1?

Can you tell me where to add it? I can't find any relavant unit tests.

Reply:
Add new test case in https://github.com/apache/seatunnel/blob/a34d25916dfcf5bcf4a9a9aadbb1856896ae7c19/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaIT.java

Reply:
> Add new test case in https://github.com/apache/seatunnel/blob/a34d25916dfcf5bcf4a9a9aadbb1856896ae7c19/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/java/org/apache/seatunnel/e2e/connector/kafka/KafkaIT.java

Thanks.